### PR TITLE
disabling tests incompatible with miri

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: changelog/2288.changed.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Disabled unit tests incompatible with Miri.
```
Comment (asomers) [inline changelog/2288.changed.md:1]: Two problems with the changelog message:
* Integration tests are affected as well as unit tests.
* You disabled the tests conditionally, not unconditionally.
```suggestion
Fix running tests with Miri.
```
You should make a similar change to the commit message, too.

**Review by asomers — CHANGES_REQUESTED** [review]
Frankly, I'm surprised that _anything_ in this crate works with Miri.  How are you invoking it, any way?

**Comment by SteveLauC** [comment]
> How are you invoking it, any way?

I am also interested in how you test Nix with miri, I tried `cargo +nightly miri test --all-features` on my host, but I am still getting errors:

```sh
$ cargo +nightly miri test --all-features
...
test test_fcntl::test_posix_fadvise::test_success ... error: unsupported operation: `open` not available when isolation is enabled
    --> /home/steve/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys/pal/unix/fs.rs:1133:36
     |
1133 |         let fd = cvt_r(|| unsafe { open64(path.as_ptr(), flags, opts.mode as c_int) })?;
     |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `open` not available when isolation is enabled
...
```

And, what is the advantage of doing so if we don't add Miri to our CI?

**Comment by devnexen** [comment]
> > How are you invoking it, any way?
> 
> I am also interested in how you test Nix with miri, I tried `cargo +nightly miri test --all-features` against your branch on my host, but I still got errors:
> 
> ```shell
> $ cargo +nightly miri test --all-features
> ...
> test test_fcntl::test_posix_fadvise::test_success ... error: unsupported operation: `open` not available when isolation is enabled
>     --> /home/steve/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys/pal/unix/fs.rs:1133:36
>      |
> 1133 |         let fd = cvt_r(|| unsafe { open64(path.as_ptr(), flags, opts.mode as c_int) })?;
>      |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `open` not available when isolation is enabled
> ...
> ```
> 
> And, what is the advantage of doing so if we don't add Miri to our CI?

Try with the following MIRIFLAGS="-Zmiri-disable-isolation -Zmiri-permissive-provenance"

**Comment by asomers** [comment]
I'm inclined not to merge this PR.  Not because it's wrong, but because I don't think that miri is of much use to a crate like this one, and the changes included in this PR add a lot of line noise.  What do you think, @devnexen ?

**Comment by devnexen** [comment]
yes true that s adding a lot of lines and we ll most likely need to care for this with new tests.

**Pull request closed without merging**